alphagov/pay-connector
# Pp 11154/add native prometheus metrics for some metrics

## WHAT YOU DID

1. Fix the metrics to use the defaultRegistry instead of a newly created one. Using the newly created one means the metrics declared manually elsewhere never appear since they appear in the defaultRegistry
2. Fix it so that prometheus metrics are exposed when running locally, before if the ECS metadata endpoint URL wasn't specified we didn't start the prometheus metrics up, and didn't expose them to /metrics
3. Add prometheus metrics for:
    1. gateway operations failures counter
    2. gateway operations response times histogram
    3. card authorisation results

Note: Please remember this is my first real java PR so please be kind :)
 
## How to test

If you run this up with pay local you can see the metrics on /metrics on the admin port (like `curl http://127.0.0.1:9301/metrics` or visit in a web browser, and you can see the metrics in there. Make a local payment `pay local payment` and you should see metric values increase

## Code review checklist

### Logging


### Documentation


* Introduced any new environment variables / removed existing environment variable
* Added new API / updated existing API definition
